bin/faucet/src/handlers.rs

```diff
@@ -51,64 +40,30 @@ pub async fn get_tokens(
     );
 
     // Check that the amount is in the asset amount options
-    if !state.faucet_config.asset_amount_options.contains(&req.asset_amount) {
+    if !state.config.asset_amount_options.contains(&req.asset_amount) {
         return Err(FaucetError::BadRequest("Invalid asset amount.".to_string()).into());
     }
 
-    let client_config = state.faucet_config.clone();
-    let mut client = build_client(client_config.database_filepath, &client_config.node_url)?;
+    let mut client = state.client.lock().await;
 
     // Receive and hex user account id
     let target_account_id = AccountId::from_hex(req.account_id.as_str())
         .map_err(|err| FaucetError::BadRequest(err.to_string()))?;
 
-    // Instantiate asset
-    let asset = FungibleAsset::new(state.id, req.asset_amount)
-        .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
-
-    // Instantiate note type
-    let note_type = if req.is_private_note {
-        NoteType::OffChain
-    } else {
-        NoteType::Public
-    };
-
-    // Instantiate transaction template
-    let tx_template = TransactionTemplate::MintFungibleAsset(asset, target_account_id, note_type);
-
-    // Instantiate transaction request
-    let tx_request = client
-        .build_transaction_request(tx_template)
-        .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
-
-    // Run transaction executor & execute transaction
-    let tx_result = client
-        .new_transaction(tx_request)
-        .map_err(|err| FaucetError::InternalServerError(err.to_string()))?;
-
-    // Get created notes from transaction result
-    let created_notes = tx_result.created_notes().clone();
+    // Execute transaction
```

Comment:
can we add some `info!` calls in between these steps (executing mint and proving + submitting)?

Reply:
Done.